app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/Save.php

```diff
@@ -169,38 +181,23 @@ public function execute()
                 $responseAjax->setError(true);
                 $responseAjax->setMessage($e->getMessage());
             } else {
-                $this->messageManager->addError($e->getMessage());
-                $this->_redirect('*/*/new', ['order_id' => $this->getRequest()->getParam('order_id')]);
+                $this->messageManager->addErrorMessage($e->getMessage());
+                return $resultRedirect->setPath('*/*/new', ['order_id' => $this->getRequest()->getParam('order_id')]);
             }
         } catch (\Exception $e) {
             $this->_objectManager->get(\Psr\Log\LoggerInterface::class)->critical($e);
             if ($isNeedCreateLabel) {
                 $responseAjax->setError(true);
                 $responseAjax->setMessage(__('An error occurred while creating shipping label.'));
             } else {
-                $this->messageManager->addError(__('Cannot save shipment.'));
-                $this->_redirect('*/*/new', ['order_id' => $this->getRequest()->getParam('order_id')]);
+                $this->messageManager->addErrorMessage(__('Cannot save shipment.'));
+                return $resultRedirect->setPath('*/*/new', ['order_id' => $this->getRequest()->getParam('order_id')]);
             }
         }
         if ($isNeedCreateLabel) {
             $this->getResponse()->representJson($responseAjax->toJson());
```

Comment:
Looks like there will be regression. Earlier redirect wasn't sent

Reply:
I'm not sure I follow. Re-check what?